File: Iteration2/code/CSE3063GRP12/src/core/enums/StudentMenu.java
package core.enums;

import core.models.concretes.Notification;
import features.login.LoginController;
import features.notification.NotificationController;
import features.student.course_registration.CourseRegistrationController;
import features.student.transcript.TranscriptController;
import features.student.weekly_schedule.WeeklyScheduleController;

public enum StudentMenu implements Menu{
    CourseRegistration, WeeklySchedule, Transcript, Notification, Logout;
    public String getItemMessage() {
        switch(this){
            case CourseRegistration:
                return ("Course Registration");
            case WeeklySchedule:
                return ("Weekly Schedule");
            case Transcript:
                return ("Transcript");
            case Notification:
                return("Notification");
            case Logout:
                return ("Logout");
            }
        return this.name();
    }

    public boolean isNotification(){
        return this == Notification;
    }
    public void navigate(){
        switch(this){
            case CourseRegistration:
                 new CourseRegistrationController();
                break;
            case WeeklySchedule:
                 new WeeklyScheduleController();
                break;
            case Transcript:
                 new TranscriptController();
                break;
             case Notification:
                new NotificationController();
                break;
            case Logout:
                new LoginController();
                break;
        }
    }
}

File: Iteration3/code/CSE3063GRP12/src/core/enums/StudentMenu.py
from core.enums.Menu import Menu
from features.student.course_registration.CourseRegistrationController import CourseRegistrationController
from features.student.transcript.TranscriptController import TranscriptController
from features.notification.NotificationController import NotificationController

class StudentMenu(Menu):
    COURSE_REGISTRATION = "Course Registration"
    WEEKLY_SCHEDULE = "Weekly Schedule"
    TRANSCRIPT = "Transcript"
    NOTIFICATION = "Notification"
    LOGOUT = "Logout"

    def get_item_message(self):
        return self.value
    
    def navigate(self):
        if self == StudentMenu.COURSE_REGISTRATION:
            CourseRegistrationController()
        elif self == StudentMenu.WEEKLY_SCHEDULE:
            from features.student.weekly_schedule.WeeklyScheduleController import WeeklyScheduleController
            WeeklyScheduleController()
        elif self == StudentMenu.TRANSCRIPT:
            TranscriptController()
        elif self == StudentMenu.NOTIFICATION:
            NotificationController()
        elif self == StudentMenu.LOGOUT:
            from features.login.LoginController import LoginController
            LoginController()    
        
    def is_notification(self):
        return self == StudentMenu.NOTIFICATION
    